Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingObserver.cs

```diff
@@ -225,80 +247,62 @@ public bool SetObserverOrigin(Vector3 origin)
         /// <param name="elapsedCookTimeSeconds">Seconds between mesh cook request and propagation of this event.</param>
         private void SurfaceObserver_OnDataReady(SurfaceData cookedData, bool outputWritten, float elapsedCookTimeSeconds)
         {
-            //We have new visuals, so we can disable and cleanup the older surface
-            GameObject surfaceToCleanup;
-            if (pendingCleanup.TryGetValue(cookedData.id.handle, out surfaceToCleanup))
+            if (outstandingMeshRequest == null)
             {
-                CleanupSurface(surfaceToCleanup);
-                pendingCleanup.Remove(cookedData.id.handle);
-            }
+                Debug.LogError(string.Format("Got OnDataReady for surface {0} while no request was outstanding.",
```

Comment:
Sounds good.  Is the idea with the Unity "Debug.Log*Format" methods that they are optimized differently at compile-time?  Or are there other benefits?